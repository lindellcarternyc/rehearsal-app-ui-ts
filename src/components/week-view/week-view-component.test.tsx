import * as React from 'react'
import * as renderer from 'react-test-renderer'
import WeekViewComponent, { WeekViewComponentProps } from './week-view-component'

const previous = jest.fn()
const next = jest.fn()
const selectDay = jest.fn()

describe('WeekViewComponent', () => {
  it('renders a week with rehearsals', () => {
    const props: WeekViewComponentProps = {
      selectDay,
      previous,
      next,
      days: [
        {
          date: 'Sun, January 14'
        },
        { 
          date: 'Mon, January 15'
        },
        {
          date: 'Tue, January 16',
          rehearsals: [
            {time: '19:00 - 22:00', material: ''}
          ]
        },
        {
          date: 'Wed, January 17'
        },
        {
          date: 'Thu, January 18'
        },
        {
          date: 'Fri, January 19'
        },
        {
          date: 'Sat, January 20',
          rehearsals: [
            {time: '13:00 - 14:30', material: ''},
            {time: '14:30 - 16:00', material: ''},
            {time: '16:00 - 18:00', material: ''}
          ]
        }
      ]
    }
    const tree = renderer.create(
      <WeekViewComponent {...props} />
    )
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with no rehearsals', () => {
    const days = [
      {
        date: 'Sun, January 14'
      },
      { 
        date: 'Mon, January 15'
      },
      {
        date: 'Tue, January 16'
      },
      {
        date: 'Wed, January 17'
      },
      {
        date: 'Thu, January 18'
      },
      {
        date: 'Fri, January 19'
      },
      {
        date: 'Sat, January 20'
      }
    ]
    const props: WeekViewComponentProps = {
      selectDay,
      previous,
      next,
      days
    }
    const tree = renderer.create(
      <WeekViewComponent {...props} />
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })
})