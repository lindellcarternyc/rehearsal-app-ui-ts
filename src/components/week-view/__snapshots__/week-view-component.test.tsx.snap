// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeekViewComponent renders a week with rehearsals 1`] = `
<div>
  <div
    className="ui fluid top fixed nine item menu"
  >
    <div
      className="icon item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="chevron left icon"
      />
      
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            S
          </div>
          <h3
            className="header"
          >
            14
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            M
          </div>
          <h3
            className="header"
          >
            15
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <a
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            T
          </div>
          <h3
            className="header"
          >
            16
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="blue circle tiny icon"
            />
          </div>
        </div>
      </div>
    </a>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            W
          </div>
          <h3
            className="header"
          >
            17
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            T
          </div>
          <h3
            className="header"
          >
            18
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            F
          </div>
          <h3
            className="header"
          >
            19
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <a
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            S
          </div>
          <h3
            className="header"
          >
            20
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="blue circle tiny icon"
            />
          </div>
        </div>
      </div>
    </a>
    <div
      className="icon item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="chevron right icon"
      />
      
    </div>
  </div>
  <div
    className="ui text container"
    style={
      Object {
        "paddingTop": "6rem",
      }
    }
  >
    <div
      className="ui celled relaxed list"
      role="list"
    >
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Sun, January 14
        </div>
        No Rehearsals
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Mon, January 15
        </div>
        No Rehearsals
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Tue, January 16
        </div>
        <div
          className="ui bulleted list"
          role="list"
        >
          <div
            className="item"
            data-value={undefined}
            onClick={[Function]}
            role="listitem"
          >
            19:00 - 22:00
          </div>
        </div>
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Wed, January 17
        </div>
        No Rehearsals
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Thu, January 18
        </div>
        No Rehearsals
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Fri, January 19
        </div>
        No Rehearsals
      </div>
      <div
        className="item"
        data-value={undefined}
        onClick={[Function]}
        role="listitem"
      >
        <div
          className="header"
        >
          Sat, January 20
        </div>
        <div
          className="ui bulleted list"
          role="list"
        >
          <div
            className="item"
            data-value={undefined}
            onClick={[Function]}
            role="listitem"
          >
            13:00 - 14:30
          </div>
          <div
            className="item"
            data-value={undefined}
            onClick={[Function]}
            role="listitem"
          >
            14:30 - 16:00
          </div>
          <div
            className="item"
            data-value={undefined}
            onClick={[Function]}
            role="listitem"
          >
            16:00 - 18:00
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`WeekViewComponent renders correctly with no rehearsals 1`] = `
<div>
  <div
    className="ui fluid top fixed nine item menu"
  >
    <div
      className="icon item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="chevron left icon"
      />
      
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            S
          </div>
          <h3
            className="header"
          >
            14
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            M
          </div>
          <h3
            className="header"
          >
            15
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            T
          </div>
          <h3
            className="header"
          >
            16
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            W
          </div>
          <h3
            className="header"
          >
            17
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            T
          </div>
          <h3
            className="header"
          >
            18
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            F
          </div>
          <h3
            className="header"
          >
            19
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="item"
      onClick={[Function]}
    >
      <div
        className="ui list"
        role="list"
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <div
          className="item"
          data-value={undefined}
          onClick={[Function]}
          role="listitem"
        >
          <div
            className="description"
          >
            S
          </div>
          <h3
            className="header"
          >
            20
          </h3>
          <div
            className="description"
          >
            <i
              aria-hidden="true"
              className="grey circle tiny inverted icon"
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="icon item"
      onClick={[Function]}
    >
      <i
        aria-hidden="true"
        className="chevron right icon"
      />
      
    </div>
  </div>
  <div
    className="ui text container"
    style={
      Object {
        "paddingTop": "6rem",
      }
    }
  >
    <p
      style={
        Object {
          "marginTop": "25%",
          "textAlign": "center",
        }
      }
    >
      No Rehearsals for 
      Jan 14
       - 
      Jan 20
    </p>
  </div>
</div>
`;
